Add files via upload

yarch3 opened the pull request.

yarch3 committed:
Add files via upload

WordyArc commented on main.dart:
```diff
@@ -0,0 +1,193 @@
+                    key: _formKey,
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                      return 'Please enter some text';
+                      }
+                      return null;
+                      },
+                    autofocus: true,
+                    decoration: InputDecoration(
+                        labelText: "What's the new goal, boss?"),
+                    onChanged: (name) {
+                      newTaskTxt = name;
+                    },
+                  )
+                ],
+              ),
+            ),
+            actions: <Widget>[
+              ElevatedButton(
+                  onPressed: () => save(),
+                  child: Text('Save'))
+            ],
+          );
+        }
+    );
+  }
+  //сохранение задачи в БД
+  void save() async {
+    if (newTaskTxt != '') {
+      Navigator.of(context).pop();
+      Task item = Task(
+          task: newTaskTxt,
+          creationTime: DateTime.now().toString()
+      );
+
+      await DB.insert(Task.destinationTable, item);
+      setState(() => newTaskTxt = '');
+      refresh();
+    }
+  }
```
Вынести всю логику работы с базой в service. Он должен только получать и доставать данные. Весь ui должен быть в компоненте

yarch3 committed:
Изменение файлов

Вынес в отдельный файл db_service, функции, работающие с бд;
вынес homepage в отдельный файл;
hp_handler - функции из homepage
